app/components/CreateCrashRecordModal.tsx

```diff
@@ -0,0 +1,377 @@
+            </Form.Control.Feedback>
+          </Form.Group>
+          {/* Crash date */}
+          <Form.Group className="mb-3">
+            <Form.Label>Crash date</Form.Label>
+            <FormControlDatePicker<CrashInputs>
+              name="crash_timestamp"
+              control={control}
+              isInvalid={Boolean(errors?.crash_timestamp)}
+              {...{ rules: { required: true } }}
+            />
+            <Form.Control.Feedback
+              type="invalid"
+              style={{
+                // manually control date picker feedback
+                display: Boolean(errors?.crash_timestamp) ? "block" : "none",
+              }}
+            >
+              Crash date is required
+            </Form.Control.Feedback>
+          </Form.Group>
+          {/* Primary address */}
+          <Form.Group className="mb-3">
+            <Form.Label>Primary address</Form.Label>
+            <Form.Control
+              {...register("rpt_street_name", {
+                required: true,
+                setValueAs: (v) => v?.trim() || null,
+              })}
+              autoComplete="off"
+              data-1p-ignore
+              placeholder="ex: W 4TH ST"
+              isInvalid={Boolean(errors.rpt_street_name)}
+            />
+            <Form.Control.Feedback type="invalid">
+              Primary address is required
+            </Form.Control.Feedback>
+          </Form.Group>
+          {/* Secondary address */}
+          <Form.Group className="mb-3">
+            <Form.Label>Secondary address</Form.Label>
+            <Form.Control
+              {...register("rpt_sec_street_name", {
+                setValueAs: (v) => v?.trim() || null,
+              })}
+              autoComplete="off"
+              data-1p-ignore
+              placeholder="ex: 100 S CONGRESS AVE"
+              isInvalid={Boolean(errors.rpt_street_name)}
+            />
+          </Form.Group>
+          {/* Repeatable unit inputs */}
+          {fields.map((field, index) => {
+            return (
+              <Card className="mb-3" key={index}>
+                <Card.Header className="d-flex justify-content-between">
+                  <span>Unit</span>
+                  {fields.length > 1 && (
+                    <Button
+                      size="sm"
+                      variant="outline-danger"
+                      onClick={() => remove(index)}
+                    >
+                      <AlignedLabel>
+                        <FaCircleMinus className="me-2" />
+                        <span>Remove</span>
+                      </AlignedLabel>
+                    </Button>
+                  )}
+                </Card.Header>
+                <Card.Body>
+                  <Form.Group className="mb-3">
+                    <Form.Label>Unit type</Form.Label>
+                    {unitTypes && (
+                      <Form.Select
+                        {...register(`units_cris.${index}.unit_desc_id`, {
```

Comment:
here you can see the path accessor pattern for managing these groups of inputs—pretty cool!

it's somehow typesafe, too 😵‍💫